rfc/rfc-78/rfc-78.md

```diff
@@ -0,0 +1,339 @@
+| Secondary indexes | supported | Not supported. Perf optimization may not kick in|
+| NBCC OR Completion time based log file ordering in a file slice | supported | Not supported. Will be using log file version and write token based ordering.|  
+
+
+### Timeline
+- Commit instants w/ completion time should be readable. HoodieInstant parsing logic to parse completion time should be ported over.
+- Commit metadata in avro instead of json should be readable.
+   - More details on this under Implementation section.
+- Pending Clustering commits using “cluster” action should be readable in 0.16.0 reader.
+- HoodieDefaultTimeline should be able to support both 0.x timeline and 1.x timeline.
+   - More details on this under Implementation section.
+- Should we port LSM reader as well to 0.16.x? Our goal here is to support snapshot, time travel and incremental queries for 1.x tables. Strictly speaking we can only do all these 3 queries in uncleaned 
+instants i.e. over instant ranges where cleaner has not been executed. So, if we guarantee that 1.x active timeline will definitely contain all the uncleaned instants, we could get away by not even porting over 
+LSM timeline reader logic to 0.16.0. 
+
+#### Completion time based read in FileGroup/FileSlice grouping and log block reads
+- As of this write up, we are not supporting completion time based log file ordering in in 0.16. Punting it as called out earlier.
+- What's the impact if not for this support:
+    - For users having single writer mode or OCC in 1.x, 0.16.0 reader not supporting completion time based read(log file ordering) should still be ok. Both 1.x reader and 0.16.0 reader should have same behavior.
+    - But only if someone has NBCC writes and have log files in different ordering written compared to the log file versions, 0.16.0 reader might result in data consistency issues. but since we are calling out that 0.16.0 is a bridge release and recommend users to migrate all the readers to 1.x fully before starting to enable any new features for 1.x tables.
+    - Example scenarios. say, we have lf1_10_25, lf2_15_20(format "logfile[index]_[starttime]_[completiontime]") for a file slice. In 1.x reader, we will order and read it as lf2 followed by lf1. w/o this support in 0.16.0, we might read lf1 followed by lf2. Just to re-iterate this might only impact users who have enabled NBCC and having multi writers writing log files in different ordering. Even if they were using OCC, one of the writers is expected to have failed (on the writer side) since data is overlapping for two writers in 1.x writer.
+
+### FileSystemView:
+- Support ignoring partially failed log files from FSV. In 0.16.0, from FSV standpoint, all log files(including partially failed) are valid. We let the log record reader ignore the partially failed log files. But
+  in 1.x, log files could be rolledback (deleted) by a concurrent rollback. So, the FSV should ensure it ignores the uncommitted log files. 
```

Comment:
reason to quote "concurrent" is: This will only impact those queries where log file was present during query planning (i.e FSV building), but it was deleted by the time query was executed (due to rollback). 
If rollback was happening serially, or if every query lines up well, where the file was never deleted in b/w a planning and an execution, we should be good. 
hope you got the gist.